users.js

```diff
@@ -413,6 +476,7 @@ class User {
 		this.inviteOnlyNextBattle = false;
 
 		// chat queue
+		/** @type {?any[]} */
```

Comment:
`@type {[string, string, Connection][]?}`

Reply:
If I do it like that, TypeScript seems to be thinking later on that its of type [string | Connection][], so later on things aren't working. In processChatQueue(), with
`let [message, roomid, connection] = this.chatQueue.shift();`
it thinks connection is of type string | Connection, but idk why it thinks that / how to fix it.